docs/design/hub-and-spoke/switch-validate/switch-validate.md

```diff
@@ -5,9 +5,680 @@
+    "burst_size": 220
+  },
+  "expected": { // only for misconfigured section
+    "flags": [
+      "STATS"
+    ],
+    "rate": 200,
+    "burst_size": 230
+  }
+}
+```
+</details>
+
+### Validate rules, meters, lags and groups installed on the switch v1
+
+`GET https://{host}/v1/switches/{switch_id}/validate`
+
+Response payload
+
+```json
+{
+  "groups": {
+    "excess": [
+      <v1GroupInfo>
+    ],
+    "misconfigured": [
+      <v1GroupInfo>
+    ],
+    "missing": [
+      <v1GroupInfo>
+    ],
+    "proper": [
+      <v1GroupInfo>
+    ]
+  },
+  "logical_ports": {
+    "error": "some_error",
+    "excess": [
+      <v1LogicalPortInfo>
+    ],
+    "misconfigured": [
+      <v1LogicalPortInfo>
+    ],
+    "missing": [
+      <v1LogicalPortInfo>
+    ],
+    "proper": [
+      <v1LogicalPortInfo>
+    ]
+  },
+  "meters": {
+    "excess": [
+      <v1MeterInfo>
+    ],
+    "misconfigured": [
+      <v1MeterInfo>
+    ],
+    "missing": [
+      <v1MeterInfo>
+    ],
+    "proper": [
+      <v1MeterInfo>
+    ]
+  },
+  "rules": {
+    "excess": [
+      -9223372036854776000
+    ],
+    "excess-hex": [
+      "800000000000001b"
+    ],
+    "misconfigured": [
+      -9223372036854776000
+    ],
+    "misconfigured-hex": [
+      "800000000000001b"
+    ],
+    "missing": [
+      -9223372036854776000
+    ],
+    "missing-hex": [
+      "800000000000001b"
+    ],
+    "proper": [
+      -9223372036854776000
+    ],
+    "proper-hex": [
+      "800000000000001b"
+    ]
+  }
+}
+```
+
+## v2 API : GET
+
+### Groups, meters, lags and rules info include flow_id, flow_path and y_flow_id(for meters,groups and rules).
+
+<details><summary>v2GroupInfo</summary>
+
+```json
+{
+  "flow_id": "10",
+  "flow_path": "e0b0716e-cc59-4ef6-8572-518bdcdc5c72",
+  "group_id": 10,
+  "excess_buckets": [
+    {
+      "port": 2,
+      "vlan": 2,
+      "vni": 2
+    }
+  ],
+  "buckets": [
+    {
+      "port": 3,
+      "vlan": 3,
+      "vni": 3
+    }
+  ],
+  "missing_buckets": [
+    {
+      "port": 4,
+      "vlan": 4,
+      "vni": 4
+    }
+  ]
+}
+```
+
+</details>
+
+<details><summary>v2LogicalPortInfo</summary>
+
+```json
+{
+  "logical_port_number": 1001,
+  "type": "lag",
+  "physical_ports": [
+    1,
+    2,
+    3
+  ]
+}
+```
+</details>
+
+<details><summary>v2MeterInfo</summary>
+
+```json
+{
+  "cookie": 10,
+  "flow_id": "10",
+  "flow_path": "e0b0716e-cc59-4ef6-8572-518bdcdc5c72",
+  "y_flow_id": "111",
+  "meter_id": 1,
+  "flags": [
+    "STATS"
+  ],
+  "rate": 300,
+  "burst_size": 200
+}
+```
+</details>
+
+<details><summary>v2RuleInfo</summary>
+
+```json
+{
+  "cookie": -9223372036854776000,
+  "cookie_hex": "800000000000001b",
+  "cookie_value": "LLDP_INPUT_PRE_DROP", // not implemented yet
+  "table_id": 10,
+  "priority": 5,
+  "flow_id": "9",
+  "flow_path": "10a1df3a-dc30-453e-8671-246f82ea0e77",
+  "y_flow_id": "111",
+  "flags": [
+    "RESET_COUNTERS"
+  ],
+  "match": [
+    {
+      "value": 1,
+      "mask": 1,
```

Comment:
one more field `is_masked` (`true`/`false`) required. Or some special value for `mask` field and some words about it here (into this document).

PS: ?
```
{
"aaa": "...",
"match": {
  "in_port": {"value": 1, "mask": 0, "is_masked": false}
}
"bbb": "...",
}
```

Reply:
will add is_masked field. 
will change to 
`"match": {
  "in_port": {"value": 1, "mask": 0, "is_masked": false}
}`